packages/common/src/utils/filters.ts

```diff
@@ -216,8 +216,14 @@ export const getFilterRuleWithDefaultValue = <T extends FilterRule>(
                             : moment();
 
                     const dateValue = valueIsDate
-                        ? formatDate(value, undefined, true)
+                        ? formatDate(
+                              // Treat the date as UTC, then remove its timezone information before formatting
+                              moment.utc(value).format('YYYY-MM-DD'),
+                              undefined,
+                              false,
```

Comment:
brought back the `false` because of the bug this value was fixing in: https://github.com/lightdash/lightdash/pull/3763